mongodb-datastore/db/mongodb_event_repo.go

```diff
@@ -0,0 +1,816 @@
+	var nextPageKey int64 = 0
+	if nextPageKeyStr != nil {
+		tmpNextPageKey, _ := strconv.Atoi(*nextPageKeyStr)
+		nextPageKey = int64(tmpNextPageKey)
+		newNextPageKey = nextPageKey + pageSize
+	} else {
+		newNextPageKey = pageSize
+	}
+
+	var sortOptions *options.FindOptions
+
+	if onlyRootEvents {
+		collectionName = collectionName + rootEventCollectionSuffix
+	}
+	if pageSize > 0 {
+		sortOptions = options.Find().SetSort(bson.D{{Key: timePropertyPath, Value: -1}}).SetSkip(nextPageKey).SetLimit(pageSize)
+	} else {
+		sortOptions = options.Find().SetSort(bson.D{{Key: timePropertyPath, Value: -1}})
+	}
+
+	collection := mr.DBConnection.Client.Database(getDatabaseName()).Collection(collectionName)
+
+	result := &EventsResult{}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	totalCount, err := collection.CountDocuments(ctx, searchOptions)
+	if err != nil {
+		logger.Error(fmt.Sprintf("error counting elements in events collection: %v", err))
+	}
+
+	cur, err := collection.Find(ctx, searchOptions, sortOptions)
+
+	if err != nil {
+		logger.Error(fmt.Sprintf("error finding elements in events collection: %v", err))
+		return nil, err
+	}
+	// close the cursor after the function has completed to avoid memory leaks
+	defer func() {
+		if err := cur.Close(ctx); err != nil {
+			logger.WithError(err).Error("Could not close cursor.")
+		}
+	}()
+	result.Events = formatEventResults(ctx, cur)
+
+	result.PageSize = pageSize
+	result.TotalCount = totalCount
+
+	if newNextPageKey < totalCount {
+		result.NextPageKey = strconv.FormatInt(newNextPageKey, 10)
+	}
+
+	return result, nil
+}
+
+func getProjectOfEvent(event models.KeptnContextExtendedCE) string {
+	collectionName := unmappedEventsCollectionName
+	// check if the data object contains the project name.
+	// if yes, store the event in the collection for the project, otherwise in /events
+	eventData, ok := event.Data.(map[string]interface{})
+	if ok && eventData["project"] != nil {
+		collectionNameStr, ok := eventData["project"].(string)
+		if ok && collectionNameStr != "" {
+			collectionName = collectionNameStr
+		}
+	}
+
+	return collectionName
+}
+
+func formatEventResults(ctx context.Context, cur *mongo.Cursor) []*models.KeptnContextExtendedCE {
+	events := []*models.KeptnContextExtendedCE{}
+	for cur.Next(ctx) {
+		var outputEvent interface{}
+		err := cur.Decode(&outputEvent)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to decode event %v", err))
+			continue
+		}
+		outputEvent, err = flattenRecursively(outputEvent)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to flatten %v", err))
+			continue
+		}
+
+		data, _ := json.Marshal(outputEvent)
+
+		var keptnEvent models.KeptnContextExtendedCE
+		err = keptnEvent.UnmarshalJSON(data)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to unmarshal %v", err))
+			continue
+		}
+
+		// backwards compatibility: transform evaluation-done events to evaluation.finished events
```

Comment:
can we delete this?

Reply:
this is still needed because there are still a couple of Keptn <= 0.7 installations out there